### DISCUSSION TITLE
Check if download incomplete using webhook

### DISCUSSION THREAD

**Discussion by altounen** [cross-seed/cross-seed#410] [Ideas]
Cross-seed correctly skips injecting a torrent if it finds a duplicate being incomplete. I'd like to use this feature to skip downloads when announces from multiple channels are very close together.

So in autobrr I'd have an external webhook that would ask cross-seed if there already is an incomplete download of the same torrent and the filter would abort. If the webhook succeeds, meaning this is the first download of this torrent, then the filter would continue and snatch the torrent and inject it normally.

**Comment by zakkarry** [comment]
So what you're proposing would require autobrr to integrate this functionality into their filtering?

Seeing as autobrr already integrates with all the clients, it might be easier to just have it done directly through brr than put the onus onto cross-seed and an external API call.

**Comment by altounen** [comment]
I don't think autobrr should necessarily be required to integrated the functionality. I just mean that since cross-seed already has this functionality internally, it could be used externally the same way autobrr already integrates with cross-seed, with a webhook.

Sure, having this functionality directly in autobrr might be ideal, but that might also be out of scope from what they're seeing the software's core functionality being, since they are referring to cross-seed for that functionality anyway.

**Comment by zakkarry** [comment]
https://github.com/cross-seed/cross-seed/pull/526
